ftdetect/uci.vim

```diff
@@ -1,2 +1,5 @@
 " UCI files are normally in /etc/config.
-autocmd BufNewFile,BufReadPost *etc/config/* setfiletype uci
+autocmd BufNewFile,BufReadPost *etc/config/*
+      \ if getline(1) . getline(2) . getline(3) =~# 'config' |
```

Comment:
```suggestion
      \ if getline(1) =~# '^\s*((c|config)|(p|package))\s' || |
      \    getline(2) =~# '^\s*((c|config)|(p|package))\s' || |
      \    getline(3) =~# '^\s*((c|config)|(p|package))\s' |
```

How about this?

This fixes the following issues:

1. concatenating the lines means "con\nfig" would match
2. a line like "# Use this file to configure nginx" would match
3. UCI does not require you to write out the whole word `config`, you can shorten it to `c`
4. It sometimes makes sense for files to start with a `package` or `p` statement rather than a `config` statement

Reference for how UCI files are parsed: https://git.openwrt.org/?p=project/uci.git;a=blob;f=file.c;h=6610f531c2f64852b032e9ead2ec30b35d61873a;hb=HEAD#l505